ethers-core/src/types/transaction/optimism_deposited.rs

```diff
@@ -0,0 +1,121 @@
+use super::rlp_opt;
+use crate::types::{Bytes, Signature, Transaction, TransactionRequest, H256, U256};
+use rlp::{Decodable, RlpStream};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+const NUM_TX_FIELDS: usize = 8;
+
+/// An error involving an OptimismDeposited transaction request.
+#[derive(Debug, Error)]
+pub enum OptimismDepositedRequestError {
+    /// When decoding a transaction request from RLP
+    #[error(transparent)]
+    DecodingError(#[from] rlp::DecoderError),
+}
+
+/// Parameters for sending a transaction
+#[derive(Clone, Default, Serialize, Deserialize, PartialEq, Eq, Debug)]
+pub struct OptimismDepositedTransactionRequest {
+    #[serde(flatten)]
+    pub tx: TransactionRequest,
+
+    /// The source hash which uniquely identifies the origin of the deposit
+    #[serde(rename = "sourceHash", skip_serializing_if = "Option::is_none")]
+    pub source_hash: Option<H256>,
+
+    /// The ETH value to mint on L2
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub mint: Option<U256>,
+
+    /// If true, the transaction does not interact with the L2 block gas pool.
+    /// Note: boolean is disabled (enforced to be false) starting from the Regolith upgrade.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub is_system_tx: Option<bool>,
+}
```

Comment:
I decided to implement this type by extending `TransactionRequest` similar to what's being done in Eip2930TransactionRequest.

Let me know if this makes sense for this use case